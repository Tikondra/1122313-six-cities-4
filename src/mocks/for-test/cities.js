export const cities = [
  {
    title: `Paris`,
    isActive: false,
    coordinates: [48.856663, 2.351556]
  },
  {
    title: `Cologne`,
    isActive: false,
    coordinates: [50.938361, 6.959974]
  },
  {
    title: `Brussels`,
    isActive: false,
    coordinates: [50.851309, 4.351718]
  },
  {
    title: `Amsterdam`,
    isActive: false,
    coordinates: [52.38333, 4.9]
  },
  {
    title: `Hamburg`,
    isActive: false,
    coordinates: [53.552645, 9.966287]
  },
  {
    title: `Dusseldorf`,
    isActive: false,
    coordinates: [51.230569, 6.787428]
  },
];
