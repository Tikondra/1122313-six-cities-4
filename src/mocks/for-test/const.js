export const SortType = {
  POPULAR: `Popular`,
  PRICE_UP: `Price: low to high`,
  PRICE_DOWN: `Price: high to low`,
  TOP_RATING: `Top rated first`
};

export const SORT_TYPES = Object.values(SortType);
