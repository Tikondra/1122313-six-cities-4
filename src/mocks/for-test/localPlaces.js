export const localPlaces = [
  {
    id: `1`,
    coordinates: [48.865610000000004, 2.350499],
    city: `Paris`,
    isPremium: false,
    isFavorite: false,
    price: 397,
    title: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `Hotel`,
    rating: 3.6,
    picture: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
    pictures: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`,
    ],
    description: [`Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`],
    room: 2,
    guests: 8,
    options: [`Air conditioning`, `Laptop friendly workspace`],
    host: {
      avatar: `/img/avatar-angelina.jpg`,
      name: `Angelina`,
      isSuper: true,
      id: 25,
    }
  }
];
