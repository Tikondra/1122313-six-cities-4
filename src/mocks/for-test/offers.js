import {TypePlace} from "../../constants";

export const offers = [
  {
    id: `bkjnlkmlkms`,
    coordinates: [48.853438, 2.349984],
    city: `Paris`,
    isPremium: true,
    price: 35,
    title: `Beautiful & luxurious apartment at great location`,
    type: TypePlace.APARTMENT,
    rating: 1.3,
    picture: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    }
  },
  {
    id: `jkbsdkjsjldk`,
    coordinates: [52.3809553943508, 4.939309666406198],
    city: `Amsterdam`,
    isPremium: false,
    price: 25,
    title: `Wood and stone place`,
    type: TypePlace.HOTEL,
    rating: 2.5,
    picture: `img/apartment-01.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    }
  },
];
