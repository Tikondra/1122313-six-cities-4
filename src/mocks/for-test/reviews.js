export const reviews = [
  {
    author: `Wilfred`,
    avatar: `img/avatar-max.jpg`,
    rating: 4,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: new Date(2020, 2, 15),
  },
  {
    author: `Alfred`,
    avatar: `img/avatar-max.jpg`,
    rating: 2,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: new Date(2020, 5, 23),
  },
  {
    author: `Tom`,
    avatar: `img/avatar-max.jpg`,
    rating: 2,
    text: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
    date: new Date(2020, 3, 23),
  },
];
