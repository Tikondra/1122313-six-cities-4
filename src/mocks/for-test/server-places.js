export const serverPlaces = [
  {
    bedrooms: 2,
    city: {
      name: `Paris`,
      location: {
        latitude: 48.85661,
        longitude: 2.351499,
        zoom: 13,
      },
    },
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    goods: [`Air conditioning`, `Laptop friendly workspace`],
    host: {
      id: 25,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      is_pro: true,
      // eslint-disable-next-line camelcase
      avatar_url: `img/avatar-angelina.jpg`
    },
    id: 1,
    images: [
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/18.jpg`,
    ],
    // eslint-disable-next-line camelcase
    is_favorite: false,
    // eslint-disable-next-line camelcase
    is_premium: false,
    location: {
      latitude: 48.865610000000004,
      longitude: 2.350499,
      zoom: 16
    },
    // eslint-disable-next-line camelcase
    max_adults: 8,
    // eslint-disable-next-line camelcase
    preview_image: `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
    price: 397,
    rating: 3.6,
    title: `Penthouse, 4-5 rooms + 5 balconies`,
    type: `hotel`
  }
];
