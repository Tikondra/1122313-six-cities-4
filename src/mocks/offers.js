import {nanoid} from "nanoid";
import {TypePlace} from "../constants";

export default [
  {
    id: nanoid(),
    coordinates: [52.3909553943508, 4.85309666406198],
    isPremium: true,
    price: 35,
    title: `Beautiful & luxurious apartment at great location`,
    type: TypePlace.APARTMENT,
    rating: 1.3,
    picture: `img/apartment-02.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 3,
    guests: 3,
    options: [`Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Clara`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.369553943508, 4.85309666406198],
    isPremium: false,
    price: 25,
    title: `Wood and stone place`,
    type: TypePlace.HOTEL,
    rating: 2.5,
    picture: `img/apartment-01.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 1,
    guests: 2,
    options: [`Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Donna`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.3909553943508, 4.929309666406198],
    isPremium: false,
    price: 30,
    title: `Nice, cozy, warm big bed apartment`,
    type: TypePlace.HOUSE,
    rating: 3.7,
    picture: `img/apartment-03.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 2,
    guests: 4,
    options: [`Towels`, `Baby seat`, `Cabel TV`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Marta`,
      isSuper: false,
    }
  },
  {
    id: nanoid(),
    coordinates: [52.3809553943508, 4.939309666406198],
    isPremium: true,
    price: 10,
    title: `Canal View Prinsengracht`,
    type: TypePlace.ROOM,
    rating: 4.1,
    picture: `img/room.jpg`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/room.jpg`, `img/apartment-01.jpg`],
    description: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    room: 4,
    guests: 1,
    options: [`Washing machine`, `Coffee machine`, `Dishwasher`, `Wi-Fi`, `Heating`, `Kitchen`, `Fridge`],
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Rose`,
      isSuper: true,
    }
  },
];
