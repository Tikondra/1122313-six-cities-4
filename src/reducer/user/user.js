import {ActionCreator as ActionCreatorApp} from "../app-state/app-state";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: null,
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
};

const ActionCreator = {
  requireAuthorization: (status, user = null) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: {status, user},
    };
  },
};

const Operation = {
  checkAuth: () => (dispatch, getState, api) => {
    return api.get(`/login`)
      .then((data) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, data.data.email));
      })
      .catch((err) => {
        throw err;
      });
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((data) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, data.data.email));
        dispatch(ActionCreatorApp.toHome());
      });
  },
};

const reducer = (state = initialState, action) => {
  if (action.type === ActionType.REQUIRED_AUTHORIZATION) {
    return Object.assign({}, state, {
      authorizationStatus: action.payload.status,
      user: action.payload.user,
    });
  }

  return state;
};

export {ActionCreator, ActionType, reducer, AuthorizationStatus, Operation};
