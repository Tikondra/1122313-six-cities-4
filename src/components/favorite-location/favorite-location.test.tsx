import * as React from "react";
import * as renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import FavoriteLocation from "./favorite-location";
import {offers} from "../../mocks/for-test/offers";
import {AuthorizationStatus} from "../../reducer/user/user";

it(`should render FavoriteLocation`, function () {
  const tree = renderer
    .create(
        <BrowserRouter>
          <FavoriteLocation
            city={offers[0].city}
            places={offers}
            status={AuthorizationStatus.AUTH}
            onClickByFavorite={jest.fn}
            onHoverPlace={jest.fn}
            onClickByHeader={jest.fn}
          />
        </BrowserRouter>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
