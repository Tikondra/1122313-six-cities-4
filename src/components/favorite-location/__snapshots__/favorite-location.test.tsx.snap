// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FavoriteLocation 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
        href="#"
      >
        <span>
          Paris
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <article
      className="favorites__card place-card"
      onMouseOver={[Function]}
    >
      <div
        className="place-card__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Beautiful & luxurious apartment at great location"
            className="place-card__image"
            height={110}
            src="img/apartment-02.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              35
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "20%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="/offer/bkjnlkmlkms"
            onClick={[Function]}
          >
            Beautiful & luxurious apartment at great location
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Apartment
        </p>
      </div>
    </article>
    <article
      className="favorites__card place-card"
      onMouseOver={[Function]}
    >
      
      <div
        className="favorites__image-wrapper place-card__image-wrapper"
      >
        <a
          href="#"
        >
          <img
            alt="Wood and stone place"
            className="place-card__image"
            height={110}
            src="img/apartment-01.jpg"
            width={150}
          />
        </a>
      </div>
      <div
        className="favorites__card-info place-card__info"
      >
        <div
          className="place-card__price-wrapper"
        >
          <div
            className="place-card__price"
          >
            <b
              className="place-card__price-value"
            >
              €
              25
            </b>
            <span
              className="place-card__price-text"
            >
              / night
            </span>
          </div>
          <button
            className="place-card__bookmark-button button place-card__bookmark-button--active"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon"
              height={19}
              width={18}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="place-card__rating rating"
        >
          <div
            className="place-card__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <h2
          className="place-card__name"
          onClick={[Function]}
        >
          <a
            href="/offer/jkbsdkjsjldk"
            onClick={[Function]}
          >
            Wood and stone place
          </a>
        </h2>
        <p
          className="place-card__type"
        >
          Hotel
        </p>
      </div>
    </article>
  </div>
</li>
`;
